src/languages/singlestoredb/singlestoredb.formatter.ts

```diff
@@ -0,0 +1,296 @@
+import Formatter from 'src/formatter/Formatter';
+import Tokenizer from 'src/lexer/Tokenizer';
+import { EOF_TOKEN, isToken, type Token, TokenType } from 'src/lexer/token';
+import { keywords } from './singlestoredb.keywords';
+import { functions } from './singlestoredb.functions';
+
+const reservedCommands = expandPhrases([
+  // queries
+  'SELECT [ALL | DISTINCT | DISTINCTROW]',
+  'FROM',
+  'WHERE',
+  'GROUP BY',
+  'HAVING',
+  'ORDER BY',
+  'LIMIT',
+  'OFFSET',
+  // Data manipulation
+  // - insert:
+  'INSERT [IGNORE] [INTO]',
+  'VALUES',
+  'REPLACE [INTO]',
+  // - update:
+  'UPDATE',
+  'SET',
+  // - delete:
+  'DELETE [FROM]',
+  // - truncate:
+  'TRUNCATE [TABLE]',
+  // Data definition
+  'CREATE [ROWSTORE] [REFERENCE | TEMPORARY | GLOBAL TEMPORARY] TABLE [IF NOT EXISTS]',
+  'CREATE [OR REPLACE] [TEMPORARY] PROCEDURE [IF NOT EXISTS]',
+  'CREATE [OR REPLACE] [EXTERNAL] FUNCTION',
+  'DROP [TEMPORARY] TABLE [IF EXISTS]',
+  // - alter table:
+  'ALTER [ONLINE] TABLE',
+  'ADD [COLUMN]',
+  'ADD [UNIQUE]',
+  'DROP [COLUMN]',
+  'MODIFY [COLUMN]',
+  'RENAME [TO | AS]',
+
+  // https://docs.singlestore.com/managed-service/en/reference/sql-reference.html
+  'ADD AGGREGATOR',
+  'ADD LEAF',
+  'AGGREGATOR SET AS MASTER',
+  'ALTER DATABASE',
+  'ALTER EVENT',
+  'ALTER FUNCTION',
+  'ALTER INSTANCE',
+  'ALTER LOGFILE GROUP',
+  'ALTER PROCEDURE',
+  'ALTER RESOURCE GROUP',
+  'ALTER SERVER',
+  'ALTER TABLESPACE',
+  'ALTER USER',
+  'ALTER VIEW',
+  'ANALYZE TABLE',
+  'ATTACH LEAF',
+  'ATTACH LEAF ALL',
+  'BACKUP DATABASE',
+  'BINLOG',
+  'BOOTSTRAP AGGREGATOR',
+  'CACHE INDEX',
+  'CALL',
+  'CHANGE',
+  'CHANGE MASTER TO',
+  'CHANGE REPLICATION FILTER',
+  'CHANGE REPLICATION SOURCE TO',
+  'CHECK BLOB CHECKSUM',
+  'CHECK TABLE',
+  'CHECKSUM TABLE',
+  'CLEAR ORPHAN DATABASES',
+  'CLONE',
+  'COMMIT',
+  'CREATE DATABASE',
+  'CREATE EVENT',
+  'CREATE INDEX',
+  'CREATE LOGFILE GROUP',
+  'CREATE PROCEDURE',
+  'CREATE RESOURCE GROUP',
+  'CREATE ROLE',
+  'CREATE SERVER',
+  'CREATE SPATIAL REFERENCE SYSTEM',
+  'CREATE TABLESPACE',
+  'CREATE TRIGGER',
+  'CREATE USER',
+  'CREATE VIEW',
+  'DEALLOCATE PREPARE',
+  'DESCRIBE',
+  'DO',
+  'DROP DATABASE',
+  'DROP EVENT',
+  'DROP FUNCTION',
+  'DROP FUNCTION',
+  'DROP INDEX',
+  'DROP LOGFILE GROUP',
+  'DROP PROCEDURE',
+  'DROP RESOURCE GROUP',
+  'DROP ROLE',
+  'DROP SERVER',
+  'DROP SPATIAL REFERENCE SYSTEM',
+  'DROP TABLESPACE',
+  'DROP TRIGGER',
+  'DROP USER',
+  'DROP VIEW',
+  'EXECUTE',
+  'EXPLAIN',
+  'FLUSH',
+  'FORCE',
+  'GRANT',
+  'HANDLER',
+  'HELP',
+  'IMPORT TABLE',
+  'INSERT',
+  'INSTALL COMPONENT',
+  'INSTALL PLUGIN',
+  'KILL',
+  'LOAD DATA',
+  'LOAD INDEX INTO CACHE',
+  'LOAD XML',
+  'LOCK INSTANCE FOR BACKUP',
+  'LOCK TABLES',
+  'MASTER_POS_WAIT',
+  'OPTIMIZE TABLE',
+  'PREPARE',
+  'PURGE BINARY LOGS',
+  'REBALANCE PARTITIONS',
+  'RELEASE SAVEPOINT',
+  'RENAME TABLE',
```

Comment:
This looks suspicious. Above we already support one syntax for renaming tables:

```sql
ALTER TABLE foo RENAME TO bar;
```

How does this other `RENAME TABLE` syntax work?

Reply:
Yeah, this is wrong, will need to fix it.